### PR TITLE
add location argument to mesh.coords

### LINKED ISSUE #5993: Better API for accessing coordinates on a `Mesh`
## ✨ Feature Request

It should be easier to access the desired coordinate within a `Mesh`, taking into account that `location` and `axis` are commonly used filters, and that these are often dynamic.

## Motivation

I'm always frustrated when I have to write code like this:

https://github.com/SciTools/iris/blob/6adac1b6f83baa2e1d4a13ced0d6748ff91092ee/lib/iris/experimental/ugrid/mesh.py#L2736-L2738

### PR DESCRIPTION
**Pull request by stephenworsley**
## 🚀 Pull Request
Closes #5993

### CODE REVIEW

File: lib/iris/experimental/ugrid/mesh.py
Comment (trexfeathers) [inline lib/iris/experimental/ugrid/mesh.py:1665]: I can understand why philosophically you are performing the validation here, but it would be cleaner to pass straight down to the underlying filtering methods. There is already some handling there:

https://github.com/SciTools/iris/blob/96cb02600b52a18ae3c30b8b145dd927d57a84fd/lib/iris/experimental/ugrid/mesh.py#L2279-L2284

Would that also offer an opportunity to avoid the duplication we currently have?

File: lib/iris/experimental/ugrid/mesh.py
Comment (trexfeathers) [inline lib/iris/experimental/ugrid/mesh.py:1683]: Let's not bother with full-on support for both 'types' of parameter. @stephenworsley and I discussed a `DeprecationWarning`, but given the impending ambition to drop the `experimental` label (#5990) I guess we would be stuck supporting the old parameters for a long time.

It's all still `experimental` right now - could we just drop `include_nodes`, `include_edges`, and `include_faces` abruptly. It's not expected to be popular API anyway. Thoughts @bjlittle @pp-mo @stephenworsley?

Comment (trexfeathers) [reply lib/iris/experimental/ugrid/mesh.py:1683]: At the very least I don't think we need to be so careful aligning possible duplicate arguments. `location` is in charge, fall back on the other arguments if present.

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Some fairly wide scope things for you. I will hold off more detailed reviewing until we have some conclusions.

Thanks!

File: lib/iris/experimental/ugrid/mesh.py
Comment (trexfeathers) [reply lib/iris/experimental/ugrid/mesh.py]: You've already made some changes but it could get a lot simpler ❤

```python
        mesh, location, axis = self.mesh, self.location, self.axis
        node_coord = mesh.coord(location="node", axis=axis)
        
        if location == "node":
            points_coord = node_coord
            bounds_connectivity = None
        else:
            points_coord = mesh.coord(location=location, axis=axis)
            bounds_connectivity = getattr(mesh, f"{location}_node_connectivity")
```

File: lib/iris/experimental/ugrid/mesh.py
Comment (trexfeathers) [reply lib/iris/experimental/ugrid/mesh.py:1635]: We discussed offline this morning and I still think it's important to push any handling down to this level, especially with the edits to `remove_coords()` as well. The code block I linked to also needs to be removed since the 'tri-state' is no longer something on offer for the user.

File: benchmarks/benchmarks/unit_style/ugrid.py
Comment (trexfeathers) [inline benchmarks/benchmarks/unit_style/ugrid.py:1]: Missed a spot:

https://github.com/SciTools/iris/blob/63c2cf25ab4a962a740a3306ffbd0ff562f48f48/benchmarks/benchmarks/unit_style/ugrid.py#L116

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (trexfeathers) [inline lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:244]: Why did this need adding? I has assumed that not passing `location` was equivalent to not passing any of the original 3 arguments?

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (trexfeathers) [inline lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:255]: ?

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (trexfeathers) [inline lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:573]: This 2D version should be testing `face`, too.

File: lib/iris/experimental/ugrid/mesh.py
Comment (trexfeathers) [inline lib/iris/experimental/ugrid/mesh.py:1]: Please could we have a What's New entry as well?

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Beautiful work @stephenworsley - neater, less confusing, and true to the original philosophy of identical signatures all eventually passing down to the `filters()` method.

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (stephenworsley) [reply lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:244]: I believe this test is set up to loop over this list of kwarg dictionaries and check that each one returns the expected output. Adding an extra dictionary here improves the coverage of this test to include the new argument.

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (stephenworsley) [reply lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:255]: As above, the addition of this argument expands the coverage of this test to check that the location arument _excludes_ appropriate coords.

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (trexfeathers) [reply lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:244]: Closing after discussion offline

File: lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py
Comment (trexfeathers) [reply lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py:255]: Agreed offline:

```suggestion
            {"location": "edge"},
```

**Review by pp-mo — APPROVED** [review]
All good I think, just needs an approval.

**Pull request merged**
